vault/cluster/inmem_layer.go

```diff
@@ -174,6 +178,7 @@ func (l *InmemLayer) clientConn(addr string) (net.Conn, error) {
 	servConn = newDelayedConn(servConn, l.readerDelay)
 
 	l.servConns[addr] = append(l.servConns[addr], servConn)
+	l.l.Unlock()
```

Comment:
Below we access `l.connectionCh` and `l.listener.pendingConns` without a lock, is this safe?

Reply:
Nope, fixed